python/lib/adaguc/runAdaguc.py

```diff
@@ -237,3 +279,60 @@ def cleanTempDir(self):
     def mkdir_p(self, directory):
         if not os.path.exists(directory):
             os.makedirs(directory)
+
+
+headers_to_skip = ["x-process-time", "age"]
+
+
+def response_to_cache(redis_pool, key, headers: str, data):
+    cacheable_headers = []
+    ttl = 0
+    for header in headers:
+        k, v = header.split(":")
+        if k not in headers_to_skip:
+            cacheable_headers.append(header)
+        if k.lower().startswith("cache-control"):
+            for term in v.split(";"):
+                if term.startswith("max-age"):
+                    try:
+                        ttl = int(term.split("=")[1])
+                    except:
+                        pass
+
+    if ttl > 0:
+        cacheable_headers_json = json.dumps(
+            cacheable_headers, ensure_ascii=True
```

Comment:
Why `ensure_ascii` (which escapes non-ascii characters, whatever that means) as we are encoding anyway?
This feel duplicate, and means the headers returned from cache can be different then the original ones provided by adaguc.